linkerd/router/src/cache.rs

```diff
@@ -0,0 +1,89 @@
+use crate::NewService;
+use ahash::AHashMap;
+use parking_lot::Mutex;
+use std::{fmt::Debug, hash::Hash, marker::PhantomData, sync::Arc};
+
+/// A [`NewService`] that lazy builds stacks for each `K`-typed key.
```

Comment:
isn't this a `NewService` that produces `Cache` `NewService`s, rather than actually lazily building stacks for `K`s?